en/payloads/README.md

```diff
@@ -0,0 +1,109 @@
+# Payloads and Cameras
+
+PX4 supports a wide range of payloads and cameras.
+
+## Mapping Drones
+
+* [Camera triggering](../peripherals/camera.md) via GPIO out
+* [Camera triggering](../peripherals/camera.md) via PWM out
+* [Camera triggering](../peripherals/camera.md) via MAVLink out
+* [Camera timing](../peripherals/camera.md#camera_capture) feedback via hotshoe input
+
+## Cargo drones and alike: Servos / Outputs
+
+* Servo or GPIO triggering (via RC or via commands)
+
+### Example Mission (in QGC)
+
+Use MAV_CMD_DO_SET_ACTUATOR to trigger one of the payload actuators.
+
+### Example script (MAVSDK)
+
+This script sends a command to set the actuator and trigger the payload release on a servo:
+
+```
+#include <mavsdk/mavsdk.h>
+#include <mavsdk/plugins/mavlink_passthrough/mavlink_passthrough.h>
+#include <mavsdk/plugins/info/info.h>
+#include <chrono>
+#include <cstdint>
+#include <iostream>
+#include <future>
+#include <memory>
+
+using namespace mavsdk;
+
+void send_actuator(MavlinkPassthrough& mavlink_passthrough,
+        float value1, float value2, float value3);
+
+int main(int argc, char **argv)
+{
+    Mavsdk mavsdk;
+    std::string connection_url;
+    ConnectionResult connection_result;
+    float value1, value2, value3;
+
+    if (argc == 5) {
+        connection_url = argv[1];
+        connection_result = mavsdk.add_any_connection(connection_url);
+        value1 = std::stof(argv[2]);
+        value2 = std::stof(argv[3]);
+        value3 = std::stof(argv[4]);
+    } 
+
+    if (connection_result != ConnectionResult::Success) {
+        std::cout << "Connection failed: " << connection_result << std::endl;
+        return 1;
+    }
+
+	bool discovered_system = false;
+    mavsdk.subscribe_on_new_system([&mavsdk, &discovered_system]() {
+		const auto system = mavsdk.systems().at(0);
+
+		if (system->is_connected()) {
+			std::cout << "Discovered system" << std::endl;
+			discovered_system = true;
+		}
+	});
+
+	std::this_thread::sleep_for(std::chrono::seconds(2));
+
+	if (!discovered_system) {
+        std::cout << "No device found, exiting." << std::endl;
+        return 1;
+    }
+
+	std::shared_ptr<System> system = mavsdk.systems().at(0);
+	for (auto& tsystem : mavsdk.systems()) {
+		auto info = Info{tsystem};
+		std::cout << info.get_identification().second.hardware_uid << std::endl;
+		if (info.get_identification().second.hardware_uid == "3762846593019032885") {
+			system = tsystem;
+		}
+	}
+
+	auto mavlink_passthrough = MavlinkPassthrough{system};
+
+	send_actuator(mavlink_passthrough, value1, value2, value3);
+
+    return 0;
+}
+
+void send_actuator(MavlinkPassthrough& mavlink_passthrough,
+		float value1, float value2, float value3)
+{
+	std::cout << "Sending message" << std::endl;
+	mavlink_message_t message;
```

Comment:
@bkueng This takes three values passed on as arguments, and sends these in [MAV_CMD_DO_SET_ACTUATOR](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_ACTUATOR).

Can you confirm a few things for me:
1. This message is now supported in PX4 master? Will this go into PX4 v1.12?
2. My memory of our discussion is that PX4 would support three assignable actuators (ie the first three assigned here) and that these would be mapped to an actual actuator/servo internally via a parameter. That actuator might be on UAVCAN or PWM or whatever. Can you confirm
   - Three params are supported for this
   - What they are named
   - What busses/outputs they can represent
   - What mitigation exists to prevent outputs being assigned that have an existing purpose - e.g. if I assigned MAIN1 as actuator 1 and that is being used as a flap.
   
@julianoes It would make sense once this is stable to remove WIP on the message and make a MAVSDK action for this. Is that on a roadmap?

Reply:
> This message is now supported in PX4 master? Will this go into PX4 v1.12?

yes

It's the 3 AUX outputs in https://docs.px4.io/master/en/concept/mixing.html#control-group-3-manual-passthrough so you can configure the mixer to assign it to any output.

Reply:
@bkueng OK, I thought those controls were for RC passthrough?
- Are you saying that they are also used for the actuator message?
- So in other words, these should already be set up for most vehicles?
- What about contention - if someone has RC and message sent via GCS will this just be "messy"?

If so, great, but does this imply anything about https://docs.px4.io/master/en/concept/custom_mixer_payload.html ? I.e. that explains how to set up a payload mixer, and it uses different control group. So when would you ever need to do this?

Reply:
> OK, I thought those controls were for RC passthrough?

It's now both. It takes the latest one that got updated/changed.

> So when would you ever need to do this?

You should almost never have to do that anymore. Except when you add a custom module that wants to control an output.